client/client.go

```diff
@@ -258,8 +259,49 @@ func (r *NotaryRepository) Initialize(rootKeyID string, serverManagedRoles ...st
 	return r.saveMetadata(serverManagesSnapshot)
 }
 
-// AddTarget adds a new target to the repository, forcing a timestamps check from TUF
-func (r *NotaryRepository) AddTarget(target *Target) error {
+// adds a TUF Change template to the given roles
+func addChange(cl *changelist.FileChangelist, c changelist.Change,
+	roles ...string) error {
+	if len(roles) == 0 {
+		roles = []string{data.CanonicalTargetsRole}
+	}
+
+	var changes []changelist.Change
+	for _, role := range roles {
+		role = strings.ToLower(role)
+
+		if !data.ValidRole(role) {
+			return data.ErrInvalidRole{Role: role}
+		}
+
+		if _, ok := data.ValidRoles[role]; ok && role != data.CanonicalTargetsRole {
```

Comment:
`ValidRole` returns true for any of the canonical roles, and also if it's a delegation role.  I wanted to exclude canonical roles that were not the target role, (see https://github.com/docker/notary/pull/371/files#diff-fce84029d346504e44323c30d21a3808R584)

But this is confusing - I think if I steal @endophage's change in [this file](https://github.com/docker/notary/pull/370/files#diff-a84805db69378bc67da7d4ae04da989aR113) this will be easier to read.